cmd/gossamer/config.go

```diff
@@ -930,3 +927,11 @@ func setDotPprofConfig(ctx *cli.Context, tomlCfg ctoml.PprofConfig, cfg *dot.Ppr
 
 	logger.Debug("pprof configuration: " + cfg.String())
 }
+
+func setStateConfig(ctx *cli.Context, tomlCfg ctoml.StateConfig, cfg *dot.StateConfig) {
+	if tomlCfg.Rewind > 0 {
+		cfg.Rewind = tomlCfg.Rewind
+	} else if ctx.GlobalIsSet(RewindFlag.Name) {
+		cfg.Rewind = ctx.GlobalUint(RewindFlag.Name)
+	}
```

Comment:
is there any reason to change the precedence here?

Reply:
The toml value was ignored previously so there is no change of precedence.
This is actually a fix I guess.

Reply:
Hm, but we should change to use the flag values with higher precedence, we might open an issue to update those places

Reply:
Done! Also regarding settings

> I would like to propose to (aka create an issue) refactor our settings to have meaningless zero values (i.e. nil for boolean, "" for strings that cannot be empty, nil for strings that can be empty, etc.). Any opinions on that?
> Also we set default values before settings flag / toml values which feels very wrong, it should be set on remaining field not specified by flag & toml